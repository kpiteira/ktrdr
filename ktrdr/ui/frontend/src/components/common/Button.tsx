import React from 'react';
import { ButtonVariant, ButtonSize } from '@/types/ui';

interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  variant?: ButtonVariant;
  size?: ButtonSize;
  isLoading?: boolean;
  isFullWidth?: boolean;
  leftIcon?: React.ReactNode;
  rightIcon?: React.ReactNode;
  children: React.ReactNode;
}

export const Button: React.FC<ButtonProps> = ({
  variant = 'primary',
  size = 'medium',
  isLoading = false,
  isFullWidth = false,
  leftIcon,
  rightIcon,
  children,
  disabled,
  className = '',
  ...rest
}) => {
  const baseClass = 'btn';
  const variantClass = `btn-${variant}`;
  const sizeClass = `btn-${size}`;
  const widthClass = isFullWidth ? 'btn-full-width' : '';
  const loadingClass = isLoading ? 'btn-loading' : '';
  
  const buttonClasses = [
    baseClass,
    variantClass,
    sizeClass,
    widthClass,
    loadingClass,
    className
  ].filter(Boolean).join(' ');

  return (
    <button
      className={buttonClasses}
      disabled={disabled || isLoading}
      {...rest}
    >
      {isLoading && (
        <span className="btn-spinner" aria-hidden="true">
          <svg width="16" height="16" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
            <circle cx="12" cy="12" r="10" fill="none" stroke="currentColor" strokeWidth="4" strokeDasharray="42" strokeDashoffset="14">
              <animateTransform 
                attributeName="transform" 
                type="rotate" 
                from="0 12 12" 
                to="360 12 12" 
                dur="1s" 
                repeatCount="indefinite" 
              />
            </circle>
          </svg>
        </span>
      )}
      {!isLoading && leftIcon && (
        <span className="btn-icon btn-icon-left">{leftIcon}</span>
      )}
      <span className="btn-text">{children}</span>
      {!isLoading && rightIcon && (
        <span className="btn-icon btn-icon-right">{rightIcon}</span>
      )}
    </button>
  );
};