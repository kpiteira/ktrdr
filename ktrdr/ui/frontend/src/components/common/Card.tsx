import React from 'react';

interface CardProps {
  title?: string;
  subtitle?: string;
  icon?: React.ReactNode;
  actions?: React.ReactNode;
  footer?: React.ReactNode;
  isLoading?: boolean;
  children: React.ReactNode;
  className?: string;
  style?: React.CSSProperties;
}

export const Card: React.FC<CardProps> = ({
  title,
  subtitle,
  icon,
  actions,
  footer,
  isLoading = false,
  children,
  className = '',
  style = {},
}) => {
  const cardClasses = [
    'card',
    isLoading ? 'card-loading' : '',
    className
  ].filter(Boolean).join(' ');

  // Apply less aggressive constraining styles to fix layout issues
  const cardStyle: React.CSSProperties = {
    ...style,
    maxWidth: '100%',
    boxSizing: 'border-box',
    // Less restrictive overflow to avoid hiding content
    overflowX: 'hidden',
    // Less aggressive containment that won't break tabs
    contain: 'paint'
  };

  return (
    <div className={cardClasses} style={cardStyle}>
      {(title || subtitle || actions) && (
        <div className="card-header">
          <div className="card-header-left">
            {icon && <div className="card-icon">{icon}</div>}
            <div className="card-titles">
              {title && <h3 className="card-title">{title}</h3>}
              {subtitle && <p className="card-subtitle">{subtitle}</p>}
            </div>
          </div>
          {actions && <div className="card-actions">{actions}</div>}
        </div>
      )}
      <div className="card-body" style={{ maxWidth: '100%', overflowX: 'hidden' }}>
        {isLoading ? (
          <div className="card-loading-spinner">
            <svg width="24" height="24" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
              <circle cx="12" cy="12" r="10" fill="none" stroke="currentColor" strokeWidth="4" strokeDasharray="42" strokeDashoffset="14">
                <animateTransform 
                  attributeName="transform" 
                  type="rotate" 
                  from="0 12 12" 
                  to="360 12 12" 
                  dur="1s" 
                  repeatCount="indefinite" 
                />
              </circle>
            </svg>
          </div>
        ) : children}
      </div>
      {footer && <div className="card-footer">{footer}</div>}
    </div>
  );
};