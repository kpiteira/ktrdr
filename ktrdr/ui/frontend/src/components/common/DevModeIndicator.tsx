import React from 'react';
import config from '@/config';

interface DevModeIndicatorProps {
  position?: 'top-right' | 'top-left' | 'bottom-right' | 'bottom-left';
}

export const DevModeIndicator: React.FC<DevModeIndicatorProps> = ({ 
  position = 'bottom-right'
}) => {
  if (!config.debug) {
    return null;
  }

  const positionClasses = {
    'top-right': 'dev-mode-indicator-tr',
    'top-left': 'dev-mode-indicator-tl',
    'bottom-right': 'dev-mode-indicator-br',
    'bottom-left': 'dev-mode-indicator-bl',
  };

  return (
    <div className={`dev-mode-indicator ${positionClasses[position]}`}>
      <div className="dev-mode-indicator-content">
        <span className="dev-mode-indicator-badge">DEV</span>
        <span className="dev-mode-indicator-version">v{config.app.version}</span>
      </div>
    </div>
  );
};