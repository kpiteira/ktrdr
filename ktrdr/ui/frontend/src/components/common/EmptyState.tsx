import React, { FC, ReactNode } from 'react';

interface EmptyStateProps {
  icon?: string;
  title: string;
  description?: string;
  action?: ReactNode;
  compact?: boolean;
}

const EmptyState: FC<EmptyStateProps> = ({
  icon = 'ðŸ“ˆ',
  title,
  description,
  action,
  compact = false
}) => {
  if (compact) {
    return (
      <div style={{
        padding: '1.5rem',
        textAlign: 'center',
        color: '#666',
        backgroundColor: '#fafafa',
        border: '1px dashed #ddd',
        borderRadius: '4px'
      }}>
        <div style={{ fontSize: '1.5rem', marginBottom: '0.5rem' }}>{icon}</div>
        <div style={{ fontSize: '0.9rem', fontWeight: '500', marginBottom: '0.25rem' }}>
          {title}
        </div>
        {description && (
          <div style={{ fontSize: '0.8rem', color: '#999' }}>
            {description}
          </div>
        )}
        {action && (
          <div style={{ marginTop: '0.75rem' }}>
            {action}
          </div>
        )}
      </div>
    );
  }

  return (
    <div style={{
      padding: '3rem 2rem',
      textAlign: 'center',
      color: '#666'
    }}>
      <div style={{ fontSize: '3rem', marginBottom: '1rem' }}>{icon}</div>
      <h3 style={{ 
        margin: '0 0 0.5rem 0', 
        color: '#333', 
        fontSize: '1.25rem',
        fontWeight: '500'
      }}>
        {title}
      </h3>
      {description && (
        <p style={{ 
          margin: '0 0 1.5rem 0', 
          fontSize: '0.9rem', 
          lineHeight: '1.4',
          color: '#666',
          maxWidth: '400px',
          marginLeft: 'auto',
          marginRight: 'auto'
        }}>
          {description}
        </p>
      )}
      {action && (
        <div>
          {action}
        </div>
      )}
    </div>
  );
};

export default EmptyState;