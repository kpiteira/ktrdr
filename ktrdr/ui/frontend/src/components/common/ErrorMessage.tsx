import React from 'react';

type ErrorSeverity = 'error' | 'warning' | 'info';

interface ErrorMessageProps {
  message: string;
  severity?: ErrorSeverity;
  details?: string;
  actionLabel?: string;
  onAction?: () => void;
  className?: string;
}

export const ErrorMessage: React.FC<ErrorMessageProps> = ({
  message,
  severity = 'error',
  details,
  actionLabel,
  onAction,
  className = '',
}) => {
  const errorClasses = [
    'error-message',
    `error-${severity}`,
    className
  ].filter(Boolean).join(' ');

  const renderIcon = () => {
    switch (severity) {
      case 'error':
        return (
          <svg width="20" height="20" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
            <path d="M12 2C6.48 2 2 6.48 2 12C2 17.52 6.48 22 12 22C17.52 22 22 17.52 22 12C22 6.48 17.52 2 12 2ZM13 17H11V15H13V17ZM13 13H11V7H13V13Z" fill="currentColor" />
          </svg>
        );
      case 'warning':
        return (
          <svg width="20" height="20" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
            <path d="M1 21H23L12 2L1 21ZM13 18H11V16H13V18ZM13 14H11V10H13V14Z" fill="currentColor" />
          </svg>
        );
      case 'info':
        return (
          <svg width="20" height="20" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
            <path d="M12 2C6.48 2 2 6.48 2 12C2 17.52 6.48 22 12 22C17.52 22 22 17.52 22 12C22 6.48 17.52 2 12 2ZM13 17H11V11H13V17ZM13 9H11V7H13V9Z" fill="currentColor" />
          </svg>
        );
      default:
        return null;
    }
  };

  return (
    <div 
      className={errorClasses} 
      role={severity === 'error' ? 'alert' : 'status'}
      aria-live={severity === 'error' ? 'assertive' : 'polite'}
    >
      <div className="error-icon">
        {renderIcon()}
      </div>
      <div className="error-content">
        <p className="error-text">{message}</p>
        {details && <p className="error-details">{details}</p>}
      </div>
      {actionLabel && onAction && (
        <button className="error-action" onClick={onAction}>
          {actionLabel}
        </button>
      )}
    </div>
  );
};