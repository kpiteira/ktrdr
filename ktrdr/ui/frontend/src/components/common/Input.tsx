import React, { forwardRef } from 'react';
import { InputType } from '@/types/ui';

interface InputProps extends Omit<React.InputHTMLAttributes<HTMLInputElement>, 'onChange'> {
  label?: string;
  type?: InputType;
  error?: string;
  helperText?: string;
  leftIcon?: React.ReactNode;
  rightIcon?: React.ReactNode;
  isFullWidth?: boolean;
  onChange?: (value: string) => void;
}

export const Input = forwardRef<HTMLInputElement, InputProps>(({
  label,
  type = 'text',
  error,
  helperText,
  leftIcon,
  rightIcon,
  isFullWidth = false,
  onChange,
  className = '',
  ...rest
}, ref) => {
  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    onChange?.(e.target.value);
  };

  const inputClasses = [
    'input-field',
    error ? 'input-error' : '',
    leftIcon ? 'has-left-icon' : '',
    rightIcon ? 'has-right-icon' : '',
    isFullWidth ? 'input-full-width' : '',
    className
  ].filter(Boolean).join(' ');

  const wrapperClasses = [
    'input-wrapper',
    isFullWidth ? 'input-wrapper-full-width' : ''
  ].filter(Boolean).join(' ');

  return (
    <div className={wrapperClasses}>
      {label && (
        <label className="input-label" htmlFor={rest.id}>
          {label}
        </label>
      )}
      <div className="input-container">
        {leftIcon && (
          <span className="input-icon input-icon-left">
            {leftIcon}
          </span>
        )}
        <input
          ref={ref}
          type={type}
          className={inputClasses}
          onChange={handleChange}
          aria-invalid={!!error}
          aria-describedby={error ? `${rest.id}-error` : helperText ? `${rest.id}-helper` : undefined}
          {...rest}
        />
        {rightIcon && (
          <span className="input-icon input-icon-right">
            {rightIcon}
          </span>
        )}
      </div>
      {error && (
        <p className="input-error-text" id={`${rest.id}-error`}>
          {error}
        </p>
      )}
      {!error && helperText && (
        <p className="input-helper-text" id={`${rest.id}-helper`}>
          {helperText}
        </p>
      )}
    </div>
  );
});