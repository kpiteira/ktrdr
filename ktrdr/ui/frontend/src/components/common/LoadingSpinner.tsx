import React, { FC } from 'react';

interface LoadingSpinnerProps {
  size?: 'small' | 'medium' | 'large';
  color?: string;
  message?: string;
  inline?: boolean;
}

const LoadingSpinner: FC<LoadingSpinnerProps> = ({
  size = 'medium',
  color = '#1976d2',
  message,
  inline = false
}) => {
  const sizeMap = {
    small: '16px',
    medium: '24px',
    large: '32px'
  };

  const spinnerSize = sizeMap[size];

  if (inline) {
    return (
      <div style={{
        display: 'inline-flex',
        alignItems: 'center',
        gap: '0.5rem'
      }}>
        <div 
          style={{
            width: spinnerSize,
            height: spinnerSize,
            border: `2px solid ${color}20`,
            borderTop: `2px solid ${color}`,
            borderRadius: '50%',
            animation: 'spin 1s linear infinite'
          }}
        />
        {message && (
          <span style={{ 
            color: '#666', 
            fontSize: size === 'small' ? '0.8rem' : '0.9rem' 
          }}>
            {message}
          </span>
        )}
        
        {/* CSS animation */}
        <style>{`
          @keyframes spin {
            0% { transform: rotate(0deg); }
            100% { transform: rotate(360deg); }
          }
        `}</style>
      </div>
    );
  }

  return (
    <div style={{
      display: 'flex',
      flexDirection: 'column',
      alignItems: 'center',
      justifyContent: 'center',
      gap: '1rem',
      padding: '2rem',
      color: '#666'
    }}>
      <div 
        style={{
          width: spinnerSize,
          height: spinnerSize,
          border: `3px solid ${color}20`,
          borderTop: `3px solid ${color}`,
          borderRadius: '50%',
          animation: 'spin 1s linear infinite'
        }}
      />
      {message && (
        <div style={{ 
          textAlign: 'center',
          fontSize: size === 'small' ? '0.8rem' : '0.9rem'
        }}>
          {message}
        </div>
      )}
      
      {/* CSS animation */}
      <style>{`
        @keyframes spin {
          0% { transform: rotate(0deg); }
          100% { transform: rotate(360deg); }
        }
      `}</style>
    </div>
  );
};

export default LoadingSpinner;