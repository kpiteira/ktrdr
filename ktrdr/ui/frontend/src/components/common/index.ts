// Export all common components
export * from './Button';
export * from './Input';
export * from './Select';
export * from './Card';
export * from './Tabs';
export * from './LoadingSpinner';
export * from './ErrorMessage';
export * from './NotificationProvider';
export * from './NotificationContainer';
export * from './DevModeIndicator';