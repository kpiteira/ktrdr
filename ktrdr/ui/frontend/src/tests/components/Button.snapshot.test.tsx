import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { Button } from '@/components/common/Button';

describe('Button Component Snapshots', () => {
  it('renders primary button correctly', () => {
    const { container } = render(<Button>Primary Button</Button>);
    expect(container).toMatchSnapshot();
  });

  it('renders secondary button correctly', () => {
    const { container } = render(<Button variant="secondary">Secondary Button</Button>);
    expect(container).toMatchSnapshot();
  });

  it('renders loading button correctly', () => {
    const { container } = render(<Button isLoading>Loading Button</Button>);
    expect(container).toMatchSnapshot();
  });

  it('renders button with icons correctly', () => {
    const { container } = render(
      <Button 
        leftIcon={<span data-testid="left-icon">←</span>}
        rightIcon={<span data-testid="right-icon">→</span>}
      >
        Button with Icons
      </Button>
    );
    expect(container).toMatchSnapshot();
  });

  it('renders disabled button correctly', () => {
    const { container } = render(<Button disabled>Disabled Button</Button>);
    expect(container).toMatchSnapshot();
  });

  it('renders full width button correctly', () => {
    const { container } = render(<Button isFullWidth>Full Width Button</Button>);
    expect(container).toMatchSnapshot();
  });
});