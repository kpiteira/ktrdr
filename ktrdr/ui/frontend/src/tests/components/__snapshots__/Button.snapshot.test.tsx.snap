// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button Component Snapshots > renders button with icons correctly 1`] = `
<div>
  <button
    class="btn btn-primary btn-medium"
  >
    <span
      class="btn-icon btn-icon-left"
    >
      <span
        data-testid="left-icon"
      >
        ←
      </span>
    </span>
    <span
      class="btn-text"
    >
      Button with Icons
    </span>
    <span
      class="btn-icon btn-icon-right"
    >
      <span
        data-testid="right-icon"
      >
        →
      </span>
    </span>
  </button>
</div>
`;

exports[`Button Component Snapshots > renders disabled button correctly 1`] = `
<div>
  <button
    class="btn btn-primary btn-medium"
    disabled=""
  >
    <span
      class="btn-text"
    >
      Disabled Button
    </span>
  </button>
</div>
`;

exports[`Button Component Snapshots > renders full width button correctly 1`] = `
<div>
  <button
    class="btn btn-primary btn-medium btn-full-width"
  >
    <span
      class="btn-text"
    >
      Full Width Button
    </span>
  </button>
</div>
`;

exports[`Button Component Snapshots > renders loading button correctly 1`] = `
<div>
  <button
    class="btn btn-primary btn-medium btn-loading"
    disabled=""
  >
    <span
      aria-hidden="true"
      class="btn-spinner"
    >
      <svg
        height="16"
        viewBox="0 0 24 24"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          fill="none"
          r="10"
          stroke="currentColor"
          stroke-dasharray="42"
          stroke-dashoffset="14"
          stroke-width="4"
        >
          <animatetransform
            attributeName="transform"
            dur="1s"
            from="0 12 12"
            repeatCount="indefinite"
            to="360 12 12"
            type="rotate"
          />
        </circle>
      </svg>
    </span>
    <span
      class="btn-text"
    >
      Loading Button
    </span>
  </button>
</div>
`;

exports[`Button Component Snapshots > renders primary button correctly 1`] = `
<div>
  <button
    class="btn btn-primary btn-medium"
  >
    <span
      class="btn-text"
    >
      Primary Button
    </span>
  </button>
</div>
`;

exports[`Button Component Snapshots > renders secondary button correctly 1`] = `
<div>
  <button
    class="btn btn-secondary btn-medium"
  >
    <span
      class="btn-text"
    >
      Secondary Button
    </span>
  </button>
</div>
`;
